packages/react/src/components/navigation/Navigation.tsx

```diff
@@ -182,7 +195,7 @@ export const Navigation = ({
 
   const [{ authenticated, navigationVariant }, dispatch] = useReducer(reducer, {
     authenticated: false,
-    navigationVariant: 'default',
+    navigationVariant: getNavigationVariantFromChild(children),
```

Comment:
Is this called before the children are rendered? I wonder if it can find the children yet here?

Reply:
On first render it is called before children are rendered for sure. But for react to find components children I can't see why it should render them before. They are basically just objects passed as children to Navigation component.